src/utils.ts

```diff
@@ -71,3 +73,131 @@ export const getCount = (item: number | string) => {
   const formattedItem = typeof item === "string" ? item.length : item;
   return wordsFormat.format(formattedItem);
 };
+
+export class Metadata {
+  private metaData: string;
+  public fileTitle: string;
+  public source: string;
+
+  constructor({
+    fileTitle,
+    transcript_by,
+    url,
+    date,
+    tags,
+    speakers,
+    categories,
+  }: MetadataProps) {
+    this.fileTitle = fileTitle;
+    this.source = url;
+
+    // eslint-disable-next-line prettier/prettier
+    this.metaData = `---\n` +
+                    `title: ${fileTitle}\n` +
+                    `transcript_by: ${transcript_by} \n`;
+
+    this.metaData += `media: ${url}\n`;
+
+    if (tags) {
+      this.metaData += `tags: ${tags}\n`;
+    }
+
+    if (speakers) {
+      this.metaData += `speakers: ${speakers}\n`;
+    }
+
+    if (categories) {
+      this.metaData += `categories: ${categories}\n`;
+    }
+    this.metaData += `date: ${date}\n`;
```

Comment:
this is not the correct date format. It should be `yyyy-md-dd` not `yyyy/mm/dd`

Reply:
alright